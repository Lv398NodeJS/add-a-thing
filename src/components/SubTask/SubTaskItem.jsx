import React, { Component } from 'react';
import {
  Button, Form, Row, Col,
} from 'react-bootstrap';
import './index.scss';

export default class Subtask extends Component {
  render() {
    const {
      subTask: { text, completed, key },
      changeSubTaskStatus,
      deleteSubTask,
    } = this.props;
    return (
      <Row className="mb-1">
        <Col className="d-flex justify-content-sm-start col-sm-11">
          <Form.Check type="checkbox" custom id={key}>
            <Form.Check.Input
              type="checkbox"
              checked={completed}
              onChange={() => changeSubTaskStatus(key)}
            />
            <Form.Check.Label className={completed && 'task-completed'}>{text}</Form.Check.Label>
          </Form.Check>
        </Col>
        <Col className="d-flex justify-content-sm-end col-sm-1">
          <Button
            variant="outline-danger"
            size="sm"
            as="input"
            value="x"
            onClick={() => deleteSubTask(key)}
          />
        </Col>
      </Row>
    );
  }
}
