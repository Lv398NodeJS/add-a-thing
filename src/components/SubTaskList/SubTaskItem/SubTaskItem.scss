$subtask-bg-onhover: linear-gradient(to right, #ececec, #f6f6f6);
$delete-button: #f6f6f6;

.subtask-row:hover {
  background: $subtask-bg-onhover;
}

.subtask-row:hover .subtask-dropdown-toggle {
  transform: scale(1);
  transition: transform 150ms ease;
}

.subtask-dropdown-toggle,
.subtask-dropdown-toggle:hover,
.subtask-dropdown-toggle:link,
.subtask-dropdown-toggle:focus,
.subtask-dropdown-toggle:active {
  background: inherit;
  transform: scale(0);
  border: none;
  outline: none;
}
.subtask-dropdown-toggle:after {
  display: none;
}
.threedots-icon {
  width: 0.9rem;
}

// Styling for checked checkbox
.custom-checkbox .custom-control-input:checked ~ .custom-control-label::before {
  background-color: var(--green);
  border-color: var(--green);
}
// Styling for label of checked checkbox
.custom-checkbox .custom-control-input:checked ~ .custom-control-label {
  color: var(--green);
  text-decoration: line-through;
}

// Styling for checked but disabled checkbox
.custom-checkbox
  .custom-control-input:checked:disabled
  ~ .custom-control-label::before {
  background-color: var(--gray);
  border-color: var(--gray);
}

// Styling for label of checked but disabled checkbox
.custom-checkbox
  .custom-control-input:checked:disabled
  ~ .custom-control-label {
  color: var(--gray);
}
