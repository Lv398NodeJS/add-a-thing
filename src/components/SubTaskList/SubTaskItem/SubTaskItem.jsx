import React, { Component } from 'react';
import {
  Button, Form, Row, Col,
} from 'react-bootstrap';
import './SubTaskItem.scss';

export default class Subtask extends Component {
  render() {
    const {
      text, completed, id, changeSubTaskStatus, deleteSubTask,
    } = this.props;

    return (
      <Row className="mb-1">
        <Col className="d-flex justify-content-sm-start col-sm-11">
          <Form.Check type="checkbox" custom id={id}>
            <Form.Check.Input
              type="checkbox"
              checked={completed}
              onChange={() => changeSubTaskStatus(id)}
            />
            <Form.Check.Label className={completed && 'subtask-completed'}>{text}</Form.Check.Label>
          </Form.Check>
        </Col>
        <Col className="d-flex justify-content-sm-end col-sm-1">
          <Button
            variant="outline-danger"
            size="sm"
            as="input"
            value="x"
            onClick={() => deleteSubTask(id)}
          />
        </Col>
      </Row>
    );
  }
}
