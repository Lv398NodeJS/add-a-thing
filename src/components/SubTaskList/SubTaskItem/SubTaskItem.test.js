import React from 'react';
import { shallow } from 'enzyme';
import SubTaskItem from './SubTaskItem';

describe('SubTaskItem component', () => {
  it('should render correctly and match the snapshot', () => {
    const subTaskItemComponent = shallow(<SubTaskItem />);
    expect(subTaskItemComponent).toMatchSnapshot();
  });

  it('should render subtask text properly', () => {
    const text = 'Subtask test text';
    const subTaskItemComponent = shallow(<SubTaskItem text={text} />);
    expect(subTaskItemComponent.find('FormCheckLabel').text()).toBe(text);
  });

  it('should render subtask checkbox and its label properly', () => {
    const completed = true;
    const subTaskItemComponent = shallow(<SubTaskItem completed={completed} />);
    expect(subTaskItemComponent.find('FormCheckInput').props().checked).toBe(completed);
    expect(subTaskItemComponent.find('FormCheckLabel').hasClass('subtask-completed')).toBe(
      completed,
    );
  });
});
