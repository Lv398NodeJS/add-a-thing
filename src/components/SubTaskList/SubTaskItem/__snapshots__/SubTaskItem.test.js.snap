// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubTaskItem component should render correctly with no props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Subtask />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef
          as="div"
          className="d-flex justify-content-sm-start col-sm-11"
        >
          <ForwardRef
            custom={true}
            disabled={false}
            inline={false}
            isInvalid={false}
            isValid={false}
            title=""
            type="checkbox"
          >
            <ForwardRef
              onChange={[Function]}
              type="checkbox"
            />
            <ForwardRef
              type="checkbox"
            />
          </ForwardRef>
        </ForwardRef>,
        <ForwardRef
          as="div"
          className="d-flex justify-content-sm-end col-sm-1"
        >
          <ForwardRef
            active={false}
            as="input"
            disabled={false}
            onClick={[Function]}
            size="sm"
            type="button"
            value="x"
            variant="outline-danger"
          />
        </ForwardRef>,
      ],
      "className": "mb-1",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": "div",
          "children": <ForwardRef
            custom={true}
            disabled={false}
            inline={false}
            isInvalid={false}
            isValid={false}
            title=""
            type="checkbox"
          >
            <ForwardRef
              onChange={[Function]}
              type="checkbox"
            />
            <ForwardRef
              type="checkbox"
            />
          </ForwardRef>,
          "className": "d-flex justify-content-sm-start col-sm-11",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <ForwardRef
                onChange={[Function]}
                type="checkbox"
              />,
              <ForwardRef
                type="checkbox"
              />,
            ],
            "custom": true,
            "disabled": false,
            "id": undefined,
            "inline": false,
            "isInvalid": false,
            "isValid": false,
            "title": "",
            "type": "checkbox",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "checked": undefined,
                "onChange": [Function],
                "type": "checkbox",
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "type": "checkbox",
                },
                "displayName": "FormCheckInput",
                "render": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": undefined,
                "className": undefined,
                "type": "checkbox",
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "type": "checkbox",
                },
                "displayName": "FormCheckLabel",
                "render": [Function],
              },
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "Input": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "type": "checkbox",
              },
              "displayName": "FormCheckInput",
              "render": [Function],
            },
            "Label": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "type": "checkbox",
              },
              "displayName": "FormCheckLabel",
              "render": [Function],
            },
            "defaultProps": Object {
              "disabled": false,
              "inline": false,
              "isInvalid": false,
              "isValid": false,
              "title": "",
              "type": "checkbox",
            },
            "displayName": "FormCheck",
            "render": [Function],
          },
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "as": "div",
          },
          "displayName": "Col",
          "render": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": "div",
          "children": <ForwardRef
            active={false}
            as="input"
            disabled={false}
            onClick={[Function]}
            size="sm"
            type="button"
            value="x"
            variant="outline-danger"
          />,
          "className": "d-flex justify-content-sm-end col-sm-1",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "active": false,
            "as": "input",
            "disabled": false,
            "onClick": [Function],
            "size": "sm",
            "type": "button",
            "value": "x",
            "variant": "outline-danger",
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "active": false,
              "disabled": false,
              "type": "button",
              "variant": "primary",
            },
            "displayName": "Button",
            "render": [Function],
          },
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "as": "div",
          },
          "displayName": "Col",
          "render": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "defaultProps": undefined,
      "displayName": "Bootstrap(Row)",
      "propTypes": undefined,
      "render": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef
            as="div"
            className="d-flex justify-content-sm-start col-sm-11"
          >
            <ForwardRef
              custom={true}
              disabled={false}
              inline={false}
              isInvalid={false}
              isValid={false}
              title=""
              type="checkbox"
            >
              <ForwardRef
                onChange={[Function]}
                type="checkbox"
              />
              <ForwardRef
                type="checkbox"
              />
            </ForwardRef>
          </ForwardRef>,
          <ForwardRef
            as="div"
            className="d-flex justify-content-sm-end col-sm-1"
          >
            <ForwardRef
              active={false}
              as="input"
              disabled={false}
              onClick={[Function]}
              size="sm"
              type="button"
              value="x"
              variant="outline-danger"
            />
          </ForwardRef>,
        ],
        "className": "mb-1",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "div",
            "children": <ForwardRef
              custom={true}
              disabled={false}
              inline={false}
              isInvalid={false}
              isValid={false}
              title=""
              type="checkbox"
            >
              <ForwardRef
                onChange={[Function]}
                type="checkbox"
              />
              <ForwardRef
                type="checkbox"
              />
            </ForwardRef>,
            "className": "d-flex justify-content-sm-start col-sm-11",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <ForwardRef
                  onChange={[Function]}
                  type="checkbox"
                />,
                <ForwardRef
                  type="checkbox"
                />,
              ],
              "custom": true,
              "disabled": false,
              "id": undefined,
              "inline": false,
              "isInvalid": false,
              "isValid": false,
              "title": "",
              "type": "checkbox",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "checked": undefined,
                  "onChange": [Function],
                  "type": "checkbox",
                },
                "ref": null,
                "rendered": null,
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "defaultProps": Object {
                    "type": "checkbox",
                  },
                  "displayName": "FormCheckInput",
                  "render": [Function],
                },
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": undefined,
                  "className": undefined,
                  "type": "checkbox",
                },
                "ref": null,
                "rendered": null,
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "defaultProps": Object {
                    "type": "checkbox",
                  },
                  "displayName": "FormCheckLabel",
                  "render": [Function],
                },
              },
            ],
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "Input": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "type": "checkbox",
                },
                "displayName": "FormCheckInput",
                "render": [Function],
              },
              "Label": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "type": "checkbox",
                },
                "displayName": "FormCheckLabel",
                "render": [Function],
              },
              "defaultProps": Object {
                "disabled": false,
                "inline": false,
                "isInvalid": false,
                "isValid": false,
                "title": "",
                "type": "checkbox",
              },
              "displayName": "FormCheck",
              "render": [Function],
            },
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "as": "div",
            },
            "displayName": "Col",
            "render": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "div",
            "children": <ForwardRef
              active={false}
              as="input"
              disabled={false}
              onClick={[Function]}
              size="sm"
              type="button"
              value="x"
              variant="outline-danger"
            />,
            "className": "d-flex justify-content-sm-end col-sm-1",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "active": false,
              "as": "input",
              "disabled": false,
              "onClick": [Function],
              "size": "sm",
              "type": "button",
              "value": "x",
              "variant": "outline-danger",
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "active": false,
                "disabled": false,
                "type": "button",
                "variant": "primary",
              },
              "displayName": "Button",
              "render": [Function],
            },
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "as": "div",
            },
            "displayName": "Col",
            "render": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": undefined,
        "displayName": "Bootstrap(Row)",
        "propTypes": undefined,
        "render": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
